includes/classes/Elasticsearch.php

```diff
@@ -1735,7 +1735,16 @@ public function add_elasticpress_version_to_user_agent( $user_agent ) {
 	 * @param array $query Query to log.
 	 */
 	protected function add_query_log( $query ) {
-		if ( ( defined( 'WP_DEBUG' ) && WP_DEBUG ) || ( defined( 'WP_EP_DEBUG' ) && WP_EP_DEBUG ) ) {
+		/**
+		 * Filter query logging. Don't log anything to the queries property when true.
+		 *
+		 * @hook ep_disable_query_logging
+		 * @param  {bool} Whether to log to the queries property. Defaults to false.
+		 * @return {bool} New value
+		 * @since  5.2.0
+		 */
+		if ( apply_filters( 'ep_disable_query_logging', false ) && ( ( defined( 'WP_DEBUG' ) && WP_DEBUG ) ||
+		( defined( 'WP_EP_DEBUG' ) && WP_EP_DEBUG ) ) ) {
```

Comment:
```suggestion
		$wp_debug      = defined( 'WP_DEBUG' ) && WP_DEBUG;
		$wp_ep_debug = defined( 'WP_EP_DEBUG' ) && WP_EP_DEBUG );
		
		/**
		 * Filter query logging. Don't log anything to the queries property when true.
		 *
		 * @hook ep_disable_query_logging
		 * @param  {bool} Whether to log to the queries property. Defaults to false.
		 * @return {bool} New value
		 * @since  5.2.0
		 */
		$disable_query_logging = apply_filters( 'ep_disable_query_logging', false )

		if ( ! $disable_query_logging && ( $wp_debug || $wp_ep_debug ) ) {
```

@rebeccahum I didn't test it, but it seems we need to negate the filter value to have it working the way we expect, no? I've expanded the code a bit to make it easier to read.

Any chance you can add a unit test too?

Reply:
Sure, that approach makes sense to me for it to override, actually yeah.